blockly-core/core/ui/contract_editor/contract_editor.js

```diff
@@ -823,6 +823,15 @@ Blockly.ContractEditor.prototype.resetExample = function (block) {
   this.testResetHandler_(block);
 };
 
+Blockly.ContractEditor.prototype.updateExampleResult = function (block, result) {
+  this.exampleViews_.forEach(function (view) {
+    if (view.isViewForBlock(block)) {
+      view.setResult(result);
+    }
+    return;
```

Comment:
Doing filter won't help anything (we still go through every element during the filter process). Doing .some could.